jbx-protocol/juice-interface
# Split lists Diff - BASE [3/3]

## What does this PR do and why?

Do not be alarmed! Most of this diff is mock data and restructuring of the `SplitsList` and `SplitsItem` components (extracts a bunch of the `SplitsItem` code into their own components). 

This PR just integrates the splits list diff on the Safe page. Next step will be adding to reconfig preview and upcoming FC card.

**e.g. 1**
<img width="1002" alt="Screen Shot 2022-12-19 at 1 16 16 pm" src="https://user-images.githubusercontent.com/96150256/208341018-58bfef5d-73ef-4147-abf9-683e60f954dc.png">

**e.g. 2**

<img width="642" alt="Screen Shot 2022-12-20 at 12 58 01 pm" src="https://user-images.githubusercontent.com/96150256/208572268-4eb9baab-df80-4511-87c5-698cc64ecbf9.png">

If you'd like to test:
1. Check out branch locally on mainnet
2. Either just go to `/@juicebox/safe?tab=history`, look through tx's

4. OR If you want to play around with your own data, replace `splits` and `oldSplits` props on `SplitsList`s instances in `ReconfigureRichPreview` with mock data (e.g. `DUMMY_SPLITS_OLD`)
5. Go to `/v2/p/233/safe`
6. Open nonce 24
7. Play around with the mock data and see if it works as you'd expect.

## Acceptance checklist
